hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/mapreduce/MapReduceBackupMergeJob.java

```diff
@@ -240,13 +240,13 @@ protected void copyMetaData(FileSystem fs, Path tmpBackupDir, Path backupDirPath
    * @throws IOException exception
    */
   protected void copyFile(FileSystem fs, Path p, Path newPath) throws IOException {
-    File f = File.createTempFile("data", "meta");
-    Path localPath = new Path(f.getAbsolutePath());
-    fs.copyToLocalFile(p, localPath);
-    fs.copyFromLocalFile(localPath, newPath);
-    boolean exists = fs.exists(newPath);
-    if (!exists) {
-      throw new IOException("Failed to copy meta file to: "+ newPath);
+    try (InputStream in = fs.open(p);
+      OutputStream out = fs.create(newPath, true)) {
+      IOUtils.copy(in, out);
+      boolean exists = fs.exists(newPath);
```

Comment:
Please move this test outside the try block, as till here, we haven't close the outout yet...

Reply:
missed this. my bad.

Fixed.